Menus/BuyMenuGUI.cpp

```diff
@@ -1884,9 +1884,10 @@ void BuyMenuGUI::CategoryChange(bool focusOnCategoryTabs)
             for (list<Entity *>::iterator oItr = catalogList[moduleID].begin(); oItr != catalogList[moduleID].end(); ++oItr)
             {
                 pSObject = dynamic_cast<SceneObject *>(*oItr);
-                // Buyable and not brain?
-                if (pSObject && pSObject->IsBuyable() && !pSObject->IsInGroup("Brains"))
-                    tempList.push_back(pSObject);
+                // Only add buyable and non-brain items, unless they are explicitly set to be available.
+				if ((pSObject && pSObject->IsBuyable() && !pSObject->IsInGroup("Brains")) || GetOwnedItemsAmount((pSObject)->GetModuleAndPresetName()) > 0 || m_AlwaysAllowedItems.find((pSObject)->GetModuleAndPresetName()) != m_AlwaysAllowedItems.end()) {
```

Comment:
Confuse.
What is happening here now?

Reply:
Weeg's "limited buy menu" logic comes after, but I figured that it should also surpass any hardcoded restrictions (Buyable and Brains), because this will allow modders to enable otherwise unbuyable items via Lua.

I did this while working on my limited ammo global script, as means to include an "ammo box" item in the buy menu only if the global script is active.